src/shillelagh/adapters/api/github.py

```diff
@@ -58,45 +59,110 @@ class Column:
-            Column("closed_at", "closed_at", StringDateTime()),
-            Column("body", "body", String()),
-            Column("author_association", "author_association", String()),
-            Column("labels", "labels[*].name", JSONString()),
-            Column("assignees", "assignees[*].login", JSONString()),
-            Column("reactions", "reactions", JSONString()),
-        ],
+        "pulls": {
+            "columns": [
+                Column("url", "html_url", String()),
+                Column("id", "id", Integer()),
+                Column("number", "number", Integer(filters=[Equal])),
+                Column("state", "state", String(filters=[Equal]), Equal("all")),
+                Column("title", "title", String()),
+                Column("userid", "user.id", Integer()),
+                Column("username", "user.login", String()),
+                Column("draft", "draft", Boolean()),
+                Column("head", "head.ref", String(filters=[Equal])),  # head.label?
+                Column("created_at", "created_at", StringDateTime()),
+                Column("updated_at", "updated_at", StringDateTime()),
+                Column("closed_at", "closed_at", StringDateTime()),
+                Column("merged_at", "merged_at", StringDateTime()),
+            ],
+            "paginated": True,
+            "processor": None,
+        },
+        "issues": {
+            "columns": [
+                Column("url", "html_url", String()),
+                Column("id", "id", Integer()),
+                Column("number", "number", Integer(filters=[Equal])),
+                Column("state", "state", String(filters=[Equal]), Equal("all")),
+                Column("title", "title", String()),
+                Column("userid", "user.id", Integer()),
+                Column("username", "user.login", String()),
+                Column("draft", "draft", Boolean()),
+                Column("locked", "locked", Boolean()),
+                Column("comments", "comments", Integer()),
+                Column("created_at", "created_at", StringDateTime()),
+                Column("updated_at", "updated_at", StringDateTime()),
+                Column("closed_at", "closed_at", StringDateTime()),
+                Column("body", "body", String()),
+                Column("author_association", "author_association", String()),
+                Column("labels", "labels[*].name", JSONString()),
+                Column("assignees", "assignees[*].login", JSONString()),
+                Column("reactions", "reactions", JSONString()),
+            ],
+            "paginated": True,
+            "processor": None,
+        },
+        "stats/punch_card": {
+            "columns": [
+                Column("dow", "$.[0]", Integer()),
+                Column("hour", "$.[1]", Integer()),
+                Column("commits", "$.[2]", Integer()),
+            ],
+            "paginated": True,
+            "processor": None,
+        },
```

Comment:
@rusackas this is what's needed to add better support for a GitHub endpoint. Here, the payload is just an array of array:

```json
[
  [0, 0, 15],
  [0, 1, 8],
  [0, 2, 6],
  [0, 3, 3],
  [0, 4, 2],
  ...
]
```

With the change we give each column a name (`dow`, `hour`, `commits`).